packages/runtime-dom/src/jsx.ts

```diff
@@ -236,10 +236,16 @@ interface AriaAttributes {
 // Vue's style normalization supports nested arrays
 export type StyleValue = string | CSSProperties | Array<StyleValue>
 
+export type ClassValue =
+  | undefined
+  | string
+  | Record<string, unknown>
```

Comment:
```suggestion
  | Record<string | number, unknown>
```

I guess number is also acceptable.